src/hooks/use-aggrid-sort.ts

```diff
@@ -8,47 +8,50 @@
 import { useCallback, useState } from 'react';
 
 export type SortConfigType = {
-    colKey: string;
-    sortWay: number;
+    colId: string;
+    sort: 'asc' | 'desc';
+    children?: boolean;
 };
 
 export type SortPropsType = {
-    onSortChanged: (colKey: string, sortWay: number) => void;
-    sortConfig: SortConfigType;
+    onSortChanged: (sortConfig: SortConfigType) => void;
+    sortConfig: SortConfigType[];
     initSort?: (colKey: string) => void;
+    children?: boolean;
 };
 
 export const SORT_WAYS = {
-    asc: 1,
-    desc: -1,
-};
-
-export const getSortValue = (sortWay: number) => {
-    if (sortWay > 0) {
-        return 'asc';
-    } else {
-        return 'desc';
-    }
+    asc: 'asc' as const,
+    desc: 'desc' as const,
 };
```

Comment:
make an enum instead to make it usable in interfaces definition

Reply:
But we can't use enum in JS code right ?

Reply:
I think you can, ComputingType enum is used in "reducer.js"

Reply:
Ah OK great, my bad